tests/src/unit-algorithms.cpp

```diff
@@ -362,4 +362,44 @@ TEST_CASE("algorithms")
         }
     }
 
+
+    SECTION("fill")
+    {
+        SECTION("fill zeros")
+        {
+            json dest_arr = {1, 1, 1, 1, 1, 1, 1, 1, 1};
+
+            std::fill(dest_arr.begin(), dest_arr.end(), 0);
+
+            CHECK(dest_arr == json({0, 0, 0, 0, 0, 0, 0, 0, 0}));
+        }
+        SECTION("fill char value")
+        {
+            json dest_arr = {1, 1, 1, 1, 1, 1, 1, 1, 1};
+            const char val = '1';
```

Comment:
Shouldn't this be a different value than used to initialize the container initially?